### DISCUSSION TITLE
Local supabase doesn't start - supabase-rest keep restarting in Docker

### DISCUSSION THREAD

**Discussion by cdedreuille** [supabase/supabase#1778] [Questions] [answered]
I'm having some issues starting a local version of Supabase with Docker. I followed the steps using the CLI but when I launch `supabase start` it gets stuck and never launches.

Looking at Docker, I can see that every container are running correctly when launching `supabase start` except `supabase-rest` who keep on trying to start, the restart on and on until getting stuck on restarting.

Is there something specific I should do?

<img width="346" alt="Screenshot 2021-05-31 at 13 49 35" src="https://user-images.githubusercontent.com/1540635/120189227-5b7eb900-c217-11eb-8b63-a9081b6acda4.png">

<img width="1038" alt="Screenshot 2021-05-31 at 13 55 58" src="https://user-images.githubusercontent.com/1540635/120189697-f081b200-c217-11eb-8be8-efa8fb190eb0.png">

**Comment by soedirgo** [comment, marked as answer]
Hey @cdedreuille, can you show the logs for supabase-rest as well as your system info?

Update: the solution is 4GB RAM + 4GB swap as per [this issue](https://github.com/supabase/cli/issues/33).

**Reply by cdedreuille** [reply]
I'm on M1. I found a similar issue here I believe: https://github.com/supabase/cli/issues/33

It doesn't show any log as it keeps restarting

<img width="1362" alt="Screenshot 2021-05-31 at 14 00 47" src="https://user-images.githubusercontent.com/1540635/120190258-a220e300-c218-11eb-85f4-240c84783ac0.png">

<img width="422" alt="Screenshot 2021-05-31 at 14 00 13" src="https://user-images.githubusercontent.com/1540635/120190178-8a495f00-c218-11eb-8c13-a8fa5d8507ff.png">

**Reply by soedirgo** [reply]
Can you try increasing Docker's RAM to 4GB? In my experience supabase-rest OOMs on Docker's 2GB default.

**Reply by cdedreuille** [reply]
It was indeed at 2GB default. I pushed it to 4GB and it stays a bit longer on start but ultimately restarts after a couple of seconds.

**Reply by cdedreuille** [reply]
Ah, it did work when I pushed the Swap to 4GB instead of 1GB.

**Comment by cdedreuille** [comment]
@soedirgo How long should it take to run? Docker says it's ready but I can still see "Starting local Supabase..." running on my terminal and when I go to http://localhost:8000/ I have this message: `{"message":"no Route matched with those values"}`.

Not sure what I'm doing wrong here.

**Reply by soedirgo** [reply]
The message at localhost:8000 is expected, it's meant to be used as `SUPABASE_URL` (we don't have the dashboard in local dev yet).

The stuck starting message is a bug from a recent commit. For now, can you try using v0.4.0? Will release a fix in a moment 🙏

**Reply by cdedreuille** [reply]
Ah got it. I was looking for a local version of the dashboard. Make sense. I guess for now I'll use another hosted project to make some tests outside of production. The CLI is pretty cool though. Thanks a lot for your help.

**Comment by fergusmeiklejohn** [comment]
Hi all, is there any update on a stable arm64 image for PostgREST? My intel Mac just died and with my new M1 I can't use the local docker supabase :-( because it crashes my app..

**Reply by soedirgo** [reply]
Latest update on this I can find is https://github.com/PostgREST/postgrest/pull/1901#issuecomment-896372323. PostgREST itself can be installed on aarch64 now, the only problem afaict is just building the image for it.
